src/Illuminate/Support/helpers.php

```diff
@@ -686,6 +686,32 @@ function last($array)
     }
 }
 
+if (! function_exists('memoize')) {
+    /**
+     * Memoize (temporarily cache) the result of a
+     * callback for the duration of the request.
+     *
+     * @param  callable $callback
+     * @return mixed
+     */
+    function memoize($callback)
+    {
+        static $caches = [];
+
+        $reflector = new ReflectionFunction($callback);
+
+        $thumbprint = $reflector->getFilename()
+            .$reflector->getStartLine()
+            .$reflector->getEndLine()
+            .serialize($reflector->getParameters())
+            .serialize($reflector->getStaticVariables());
+
+        $hash = md5($thumbprint);
```

Comment:
`md5` has been voted to be entitely removed without deprecated warnings on https://wiki.php.net/rfc/release-md5-deprecation

Reply:
@deleugpn The proposal of that RFC is to remove MD5 checksums for PHP downloads, not to remove the PHP `md5()` function.

Reply:
That's a shame 😞